openfe/setup/molecule.py

```diff
@@ -0,0 +1,30 @@
+# This code is part of OpenFE and is licensed under the MIT license.
+# For details, see https://github.com/OpenFreeEnergy/openfe
+
+from openfe.utils.molhashing import hashmol
+
+
+class Molecule:
+    """Molecule wrapper to provide proper hashing and equality.
+
+    Parameters
+    ----------
+    rdkit : rdkit.Mol
+        rdkit representation of the molecule
+    """
+    def __init__(self, rdkit):
+        self._rdkit = rdkit
+        self._hash = None
```

Comment:
Is there a case where we don't want this to be calculated on init?

Reply:
I was 50/50 on which way to go with this. The delayed calculation idiom is more a matter of autopilot than any opinion; happy to switch it if that's preferred.

I suppose that delayed calculation is slower by the `if` statement, so if we always calculate it anyway, might as well do it in `__init__`. But I really doubt it'll be a bottleneck either way.

Reply:
Yeah I guess it depends if we always expect our use cases to use the hash? If we have cases where it's not needed then this approach makes more sense.

I don't have a strong opinion either way to be honest, the `if` is going to be super inconsequential compared to the cost of `MolToSmiles` (although even then that's pretty quick).

Reply:
put this ahead of time, it reduces the boilerplate and we'll likely always use it